src/component/HomepageBody.jsx

```diff
@@ -0,0 +1,95 @@
+import { Categories, Img } from "./images"
+import { useCategoriesContext } from "../context/categories-context"
+import {ThreeDots} from "react-loader-spinner";
+import {Link} from "react-router-dom";
+
+
+
+export const HomepageBody = () => {
+
+const {categories} = useCategoriesContext();
+if(categories === 'loading')
+{
+return (<h1>
+  <ThreeDots color="#2BAD60" height="100" width="100" />
+</h1>
+)
+}
+else{
```

Comment:
suggestion: some issue with indentation. Please look into it

Reply:
Changes Done